wp-content/themes/workingnyc/src/js/modules/programs-archive.js

```diff
@@ -187,7 +227,179 @@ export default {
+
+      if(this.headers.pages<=6){
+        this.firstPage = false;
+        this.lastPage = false;
+        this.totalPages = this.headers.pages;
+      }
+      else{
+        this.firstPage = true;
+        this.lastPage = true;
+        this.totalPages = [2,3,4,5]
+      } 
+    },
+
+    /**
+     * Fetch the posts of updated page
+     */
+    updatePagination: function(){
+      this.scrollToTop();
+      let url = [
+        this.domain,
+        this.lang.path,
+        this.endpoints[this.type],
+        this.buildUrlQuery(this.query)
+      ].join('');
+
+      fetch(url)
+      .then(this.response)
+      .then(data => {
+        let headers = Object.assign({}, this.headers);
+        this.posts.length=1;
+        let currentPost = {
+          'posts': data,
+          'headers': headers,
+          'query': this.query,
+          'show': true
+        };
+        this.posts.push(currentPost);
+        this.setPagination();      
+      }).catch(this.error);
+    },
+
+    previousPage:function(event){
+      let currPage = this.posts[1].query.page;
+      this.$set(this.query, 'page', currPage-1);
+      this.updatePagination();
+    },
+
+    nextPagination:function(event){
+      let currPage = this.posts[1].query.page;
+      this.$set(this.query, 'page', currPage+1);
+      this.updatePagination();
+    },
+
+    immediatePage: function(event){
+      let change = parseInt(event.target.dataset.amount);
+      this.$set(this.query, 'page', change);
+      this.updatePagination();
+    },
+
+    updateQuery: function(taxonomy, terms) {
+      return new Promise((resolve) => { // eslint-disable-line no-undef
+        this.$set(this.query, taxonomy, terms);
+        this.$set(this.query, 'page', 1);
+        // hide all of the posts
+        if(this.posts.length>1){
+          this.posts.length=1;
+        }
+        resolve();
+      })
+      .then(this.wp)
+      .catch(message => {
+      });
+    },
+
+    /**
+     * Update the pagination array
+     * The maximum pages: 6
+     */
+    setPagination: function(){
+
+      /**
+       * If number of pages is less than 7
+       *  The totalPages = [1,2,3,4,5,6]
+       *  The Pagination = 1,2,3,4,5,6
+       * else
+       *  The totalPages = [2,3,4,5]
+       *  The Pagination = firstPage,2,3,4,5..lastPage
+       */
+
+      if(this.headers.pages<=6){
+        this.firstPage = false;
+        this.lastPage = false;
+        this.totalPages = this.headers.pages;
+      }
+      else{
+        this.firstPage = true;
+        this.lastPage = true;
+
+        //If current page is between 1 and 4, the totalPages = [2,3,4,5]
+        if(this.query.page<=4){
+          this.totalPages = [2,3,4,5]
+        }
+        else{
+          /**
+           * If the current page is greater than 4, we have to show the previous and next page number
+           */ 
+          if((this.query.page!=this.totalPages[1])&&this.headers.pages-this.query.page>4){
+            let tempPages=[4];
```

Comment:
```suggestion
            let tempPages=[];
```
Minor - it's unclear why the `4` is necessary

Reply:
I wanted to limit the size of array to 4 as the maximum number of middle pages that we show with the users would be 4.

Reply:
Can you make this change? (here and on line 345)